docs/core/extensions/custom-configuration-provider.md

```diff
@@ -0,0 +1,63 @@
+---
+title: Implement a custom configuration provider in .NET
+description: Learn how to implement a custom configuration provider in .NET applications.
+author: IEvangelist
+ms.author: dapine
+ms.date: 09/15/2020
+ms.topic: how-to
+---
+
+# Implement a custom configuration provider in .NET
+
+There are many [configuration providers](configuration-providers.md) available for common configuration sources such as INI, JSON, and XML files. You may need to implement a custom configuration provider when one of the available providers doesn't suit your application needs. In this article, you'll learn how to implement a custom configuration provider that relies on a database as its configuration source.
+
+## Custom configuration provider
+
+The sample app demonstrates how to create a basic configuration provider that reads configuration key-value pairs from a database using [Entity Framework (EF) Core](/ef/core).
+
+The provider has the following characteristics:
+
+- The EF in-memory database is used for demonstration purposes. To use a database that requires a connection string, implement a secondary `ConfigurationBuilder` to supply the connection string from another configuration provider.
+- The provider reads a database table into configuration at startup. The provider doesn't query the database on a per-key basis.
+- Reload-on-change isn't implemented, so updating the database after the app starts has no effect on the app's configuration.
+
+Define a `Settings` record type entity for storing configuration values in the database.
+
+*Models/Settings.cs*:
```

Comment:
I'm not a fan of this format for code samples. It looks a bit odd to me. How about something like this:

```suggestion
Define a `Settings` record type entity for storing configuration values in the database. For example, you could add a *Settings.cs* file in your *Models* folder:
```